lib/spack/spack/fetch_strategy.py

```diff
@@ -776,6 +776,10 @@ def git(self):
         if not self._git:
             self._git = which('git', required=True)
 
+            # Disable advice for a quieter fetch
+            self._git.add_default_arg('-c')
+            self._git.add_default_arg('advice.detachedHead=false')
```

Comment:
See comment at https://github.com/spack/spack/pull/21233#discussion_r564034458 -- if e.g. there was some more general system to configure output from spack messages, subprocesses, and other outputs, this could be a configuration instead of code change, potentially. Let me know if there's any value to that idea and I can make an issue with my thoughts!

Reply:
I agree that it would be cool for Spack to ship its own `.gitconfig` that is used for all git fetching. This would allow us to disable all advice, not just a single one.